src/plugins/dashboard/public/application/top_nav/show_share_modal.tsx

```diff
@@ -27,6 +28,14 @@ interface ShowShareModalProps {
   dashboardStateManager: DashboardStateManager;
 }
 
+export const showPublicUrlSwitch = (anonymousUserCapabilities: Capabilities) => {
+  if (!anonymousUserCapabilities.dashboard) return false;
+
+  const dashboard = (anonymousUserCapabilities.dashboard as unknown) as DashboardCapabilities;
```

Comment:
nit: This kind of TS casting is usually a code smell.

Reply:
@clintandrewhall how would you write this bit?

Reply:
Capabilities are very weakly typed, so this type of code is unfortunately somewhat common - it's a limitation of the original design, and the fact that the implementation is split between OSS and X-Pack.

Reply:
It is not ideal, but I will live it as-is unless somebody has a nicer way of typing this.